### PR TITLE
Add a deprecation notice for legacy editor

### LINKED ISSUE #5223: Add a modal showing the message of deprecation for legacy editor
## Description

I noticed few bugs and additional features required in legacy editor and thought of working on it. 
When I opened the issue, one of the mentors mentioned that the legacy editor is slowly being deprecated.
https://github.com/publiclab/plots2/issues/5193#issuecomment-474905205

## Expected
In case a newbie starts working on some bugs or new feature of that editor, we can show in the modal showing the message of deprecation.
And for users, It would be shown to them so that they get accustomed to the new editors and when the legacy editor goes away they don't face problem shifting to the editor
So It's better to prefer rich/markdown editor.

### PR DESCRIPTION
**Pull request by CleverFool77**
Fixes #5223 (<=== Add issue number here)

I added deprecation notice

Screenshot from 2019-04-15 21-27-12

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by dinaelhanan** [comment]
@CleverFool77 you may have to restart the build because it's saying it's an internal server error on travis' side. See [this](https://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit)

**Comment by CleverFool77** [comment]
Thank you so much for the information @dinaelhanan. I've been trying to find the solution of this since a day.
I'll try the method given in link

**Comment by SidharthBansal** [comment]
@CleverFool77 do you need any help?

**Comment by CleverFool77** [comment]
@SidharthBansal This PR is complete. Please Review ?

File: app/controllers/editor_controller.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class EditorController < ApplicationController
   def legacy
     # /post/?i=http://myurl.com/image.jpg
     flash.now[:notice] = "This is the legacy editor. For the new rich editor, <a href='/editor'>click here</a>."
+    flash.now[:error] = "Deprecation notice: Legacy editor will be discontinued soon, please use rich/markdown editor."
```
Comment (gauravano) [inline app/controllers/editor_controller.rb]:
```suggestion
    flash.now[:warning] = "Deprecation notice: Legacy editor will be discontinued soon, please use rich/markdown editor."
```

How about using warning alert as error seems a bit more? What do you think @CleverFool77? Thanks!

**Comment by gauravano** [comment]
Also @CleverFool77, you can add `review-me` label to the PR when your PR gets ready to review as sometimes it becomes tough to go through every PR but this way we can fasten the pushing of those PRs which are near completion.

Thanks!

File: app/controllers/editor_controller.rb (hunk 1 above)
Comment (CleverFool77) [reply app/controllers/editor_controller.rb]: Hi @gauravano 
I've updated The PR as requested. Thanks !!!

**Comment by CleverFool77** [comment]
Hi @gauravano I'll keep in mind. I did add some `review-me` labels right now to right some the PRs. Thanks !!
> Also @CleverFool77, you can add `review-me` label to the PR when your PR gets ready to review as sometimes it becomes tough to go through every PR but this way we can fasten the pushing of those PRs which are near completion.
> 
> Thanks!

**Comment by gauravano** [comment]
Please update the screen shot. Thanks!

**Comment by CleverFool77** [comment]
Hi @gauravano 
Here is the updated screenshot
Thanks !!

Screenshot from 2019-04-15 21-27-12

**Comment by gauravano** [comment]
@publiclab/community-reps fine?

**Comment by SidharthBansal** [comment]
Perfect!!!

**Pull request merged**